### ISSUE TITLE
[Feature]: Add compression to Object3D (Point Clouds)

### ISSUE THREAD

**Issue by mokrueger** [wandb/wandb#9009]
### Description

Hi there I am currently working on a project that involves lots of point cloud data. I noticed that the current implementation for Point Clouds a rather simple json format to store the data.

The following is for the case of using the from_numpy function:
```python
            # file: object_3d.py
            tmp_path = os.path.join(MEDIA_TMP.name, runid.generate_id() + ".pts.json")
            with codecs.open(tmp_path, "w", encoding="utf-8") as fp:
                json.dump(
                    data,
                    fp,
                    separators=(",", ":"),
                    sort_keys=True,
                    indent=4,
                )
            self._set_file(tmp_path, is_tmp=True, extension=".pts.json")
```

For my purposes this yields file sizes >50Mb which really slows down the visualizations as it takes a while to load. This could easily be fixed within python by adding an optional compression argument. However, I am unsure how the front-end would implement this. 

### Suggested Solution

For my example using bz2 compression of the json data would yield a file size of 7.6MB (53.6MB original). 
Gzip yields around 10MB.
For reference, compressing the raw numpy data (so no json formatting) could achieve the smallest compression 4,1MB

Any sort of compression would probably be beneficial since the Json / whatever format should not be required to be human readable.

Kind regards ;)

**Comment by paulosabile-wb** [comment]
Hey @mokrueger Thanks for sharing this as well. Let me also review this and get back to you.

When you mentioned that file sized that are greater than 5 mb slows down the visualizations in the UI, could you please tell us how long did it take for the workspace to render the object that you were trying to log?

It will also help if you can share more context about the experiment that you are trying to setup here so we can further review and identify the next steps. Thanks!

**Comment by fmamberti-wandb** [comment]
Hi @mokrueger - thank you for sharing the details for your use case and a potential solution. I have now raised this as a feature request for our product team and will keep you posted with any updates.

**Comment by kabouzeid** [comment]
Indeed, the current file format is HUGE. Numbers are stored in text form.

**Comment by karimknaebel** [comment]
@fmamberti-wandb are there any updates on this?
